lightning/src/ln/channelmonitor.rs

```diff
@@ -468,12 +480,11 @@ pub struct ChannelMonitor {
 	to_remote_rescue: Option<(Script, SecretKey)>,
 
 	// Used to track outpoint in the process of being claimed by our transactions. We need to scan all transactions
-	// for inputs spending this. If height timer (u32) is expired and claim tx hasn't reached enough confirmations
-	// before, use TxMaterial to regenerate a new claim tx with a satoshis-per-1000-weight-units higher than last
-	// one (u64), if timelock expiration (u32) is near, decrease height timer, the in-between bumps delay.
-	// Last field cached (u32) is height of outpoint confirmation, which is needed to flush this tracker
-	// in case of reorgs, given block timer are scaled on timer expiration we can't deduce from it original height.
-	our_claim_txn_waiting_first_conf: HashMap<BitcoinOutPoint, (u32, TxMaterial, u64, u32, u32)>,
+	// for inputs spending this. If timelocks are far in the future, it's worthy to aggregate inputs claims with same
+	// timelock expiration scale in one claiming tx to save on fees. If this tx doesn't confirm before height timer
+	// we need to bump it (RFB or CPFP). If an input has been part of an aggregate tx at first claim try, we need to
+	// keep it within another bumped aggregate tx to comply with RBF rules.
+	our_claim_txn_waiting_first_conf: HashMap<BitcoinOutPoint, ClaimTxBumpMaterial>,
```

Comment:
Hmm? I seem to be misunderstanding this - this map is from a spent outpoint to the info on the tx that spent it, but then inside it is a map to the list of other inputs being spent, but those should also go in this map? This seems really likely to end up inconsistent. Can we instead index this by the txid we're waiting on confirmation of (or original txid and a map from bumped txids to the original one), plus a map from outpoints to the original txid? This would remove the need for per_input_material being indexed by the input vout so that it could just be by the input in the transaction, and also would let us create single txn that spend multiple txes (eg one claim tx spending htlc txn plus commitment txn).